src/Model/ReviewedPreprint.php

```diff
@@ -0,0 +1,219 @@
+        $this->status = $status;
+        $this->stage = $stage;
+        $this->doi = $doi;
+        $this->indexContent = $indexContent;
+        $this->authorLine = $authorLine;
+        $this->titlePrefix = $titlePrefix;
+        $this->published = $published;
+        $this->reviewedDate = $reviewedDate;
+        $this->statusDate = $statusDate;
+        $this->volume = $volume;
+        $this->elocationId = $elocationId;
+        $this->pdf = $pdf;
+        $this->subjects = $subjects;
+        $this->curationLabels = $curationLabels;
+        $this->image = $image;
+    }
+
+    final public function getIdentifier(): Identifier
+    {
+        return Identifier::reviewedPreprint($this->id);
+    }
+
+    /**
+     * @return string|null
+     */
+    public function getAuthorLine()
+    {
+        return $this->authorLine;
+    }
+
+    /**
+     * @return Sequence|null
+     */
+    public function getCurationLabels()
+    {
+        return $this->curationLabels;
+    }
+
+    /**
+     * @return string|null
+     */
+    public function getDoi()
+    {
+        return $this->doi;
+    }
+
+    /**
+     * @return string|null
+     */
+    public function getElocationId()
+    {
+        return $this->elocationId;
+    }
+
+    /**
+     * @return string
+     */
+    public function getId()
+    {
+        return $this->id;
+    }
+
+    /**
+     * @return string|null
+     */
+    public function getIndexContent()
+    {
+        return $this->indexContent;
+    }
+
+    /**
+     * @return DateTimeImmutable|null
+     */
+    public function getPublishedDate()
+    {
+        return $this->published;
+    }
+
+    /**
+     * @return string|null
+     */
+    public function getPdf()
+    {
+        return $this->pdf;
+    }
+
+    /**
+     * @return DateTimeImmutable|null
+     */
+    public function getReviewedDate()
+    {
+        return $this->reviewedDate;
+    }
+
+    /**
+     * @return string
+     */
+    public function getStage()
+    {
+        return $this->stage;
+    }
+
+    /**
+     * @return string
+     */
+    public function getStatus()
+    {
+        return $this->status;
+    }
+
+    /**
+     * @return Sequence|null
+     */
+    public function getSubjects()
+    {
+        return $this->subjects;
+    }
+
+    /**
+     * @return DateTimeImmutable|null
+     */
+    public function getStatusDate()
+    {
+        return $this->statusDate;
+    }
+
+    /**
+     * @return string|null
+     */
+    public function getTitlePrefix()
+    {
+        return $this->titlePrefix;
+    }
+
+    /**
+     * @return string
+     */
+    public function getTitle()
+    {
+        return $this->title;
+    }
+
+    /**
+     * @return Image|null
+     */
+    public function getImage()
+    {
+        return $this->image;
+    }
+
+    /**
+     * @return int|null
+     */
+    public function getVolume()
+    {
+        return $this->volume;
+    }
+
+    public function getThumbnail()
```

Comment:
@nlisgo we need ReviewedPreprint model to implement `HasThumbnail` in order to be able to use it in journal. Because `CreateTeaserImage` trait on Journal has a method called `smallTeaserImage` which accept a model that implements the `HasThumbnail` interface.

Reply:
@nlisgo Can we say this is resolved?